src/py21cmfast/lightconers.py

```diff
@@ -161,7 +160,9 @@ def with_equal_cdist_slices(
         )
 
     def make_lightcone_slices(
-        self, c1: Coeval, c2: Coeval
+        self,
+        c1: Coeval,  # noqa: F821
+        c2: Coeval,  # noqa: F821
```

Comment:
is it better to have the unused import or the undefined typehints ehre?

Reply:
I would use a "string" to forward reference the type, but the ruff "upgrade" format rule keeps automatically taking away the string, which then results in F821 firing. This is the best I could manage. See https://github.com/astral-sh/ruff/issues/8700